misc-scripts/assembly/ensembl_assembly_to_chain.pl

```diff
@@ -160,7 +160,18 @@ sub run_on_dba {
 sub get_liftover_mappings {
   my ($core_dba) = @_;
   my $mappings = $core_dba->get_MetaContainer()->list_value_by_key('liftover.mapping');
-  return [ map { $_ =~ /.+:(.+)#.+:(.+)/; [$1, $2] } @{$mappings} ];
+  my %unique_mappings;
+  for (@$mappings){
+      die "Can't parse mapping string for coord system version '$_'!\n"
+          unless /.+:(.+)#.+:(.+)/;
+      my $version_1 = $1;
+      my $version_2 = $2;
+      $unique_mappings{$version_1. ":". $version_2}++;
+  }
+  my @unique_mappings;
+  push @unique_mappings, [split(/:/, $_)]
```

Comment:
Quickest way I could think of to make a unique list of pairs... hacky I
know.

On Thu, Feb 4, 2016 at 2:22 PM, Kieron Taylor <notifications@github.com>
wrote:

> In misc-scripts/assembly/ensembl_assembly_to_chain.pl
> <https://github.com/Ensembl/ensembl/pull/117#discussion_r51877825>:
>
> > @@ -160,7 +160,18 @@ sub run_on_dba {
> >  sub get_liftover_mappings {
> >    my ($core_dba) = @_;
> >    my $mappings = $core_dba->get_MetaContainer()->list_value_by_key('liftover.mapping');
> > -  return [ map { $_ =~ /.+:(.+)#.+:(.+)/; [$1, $2] } @{$mappings} ];
> > +  my %unique_mappings;
> > +  for (@$mappings){
> > +      die "Can't parse mapping string for coord system version '$_'!\n"
> > +          unless /.+:(.+)#.+:(.+)/;
> > +      my $version_1 = $1;
> > +      my $version_2 = $2;
> > +      $unique_mappings{$version_1. ":". $version_2}++;
> > +  }
> > +  my @unique_mappings;
> > +  push @unique_mappings, [split(/:/, $_)]
>
> Seems a little odd to join the versions together and then split them up
> again.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ensembl/ensembl/pull/117/files#r51877825>.
>